Redefine PTRDIFF_TYPE

Change of SIZE_TYPE from "long unsigned int" to "unsigned int" introduced
regression due to existing PTRDIFF_TYPE.

Now to fix regression convert PTRDIFF_TYPE to simple "int".

Fix is taken from current development branch of GCC for ARC and will be a
part of the next release of ARC tools, so at that point patch should be dropped.

https://github.com/foss-for-synopsys-dwc-arc-processors/gcc/commit/846e92167aa4f486259c9ff44bb4e6cce6097fa4

Signed-off-by: Claudiu Zissulescu <claziss@synopsys.com>
---

--- a/gcc/config/arc/arc.h
+++ b/gcc/config/arc/arc.h
@@ -488,7 +488,7 @@ if (GET_MODE_CLASS (MODE) == MODE_INT		\
 #define DEFAULT_SIGNED_CHAR 0
 
 #define SIZE_TYPE "unsigned int"
-#define PTRDIFF_TYPE "long int"
+#define PTRDIFF_TYPE "int"
 #define WCHAR_TYPE "int"
 #define WCHAR_TYPE_SIZE 32
 
